src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -66,32 +70,101 @@ export default () => {
             // applied in their correct and specific order. If this queue was not paused, then there would be a lot of
             // onyx data being applied while we are fetching the missing updates and that would put them all out of order.
             SequentialQueue.pause();
-            let canUnpauseQueuePromise;
+            let canUnpauseQueuePromise: Promise<Response | void>;
 
             // The flow below is setting the promise to a reconnect app to address flow (1) explained above.
             if (!lastUpdateIDAppliedToClient) {
                 Log.info('Client has not gotten reliable updates before so reconnecting the app to start the process');
 
+                deferredUpdateBeforeReconnect = updateParams;
+
                 // Since this is a full reconnectApp, we'll not apply the updates we received - those will come in the reconnect app request.
                 canUnpauseQueuePromise = App.finalReconnectAppAfterActivatingReliableUpdates();
             } else {
                 // The flow below is setting the promise to a getMissingOnyxUpdates to address flow (2) explained above.
+
+                // Get the number of deferred updates before adding the new one
+                const existingDeferredUpdatesCount = deferredUpdates.length;
+                // Add the new update to the deferred updates
+                deferredUpdates.push(updateParams);
```

Comment:
I think it's better if you have this as a keyValue object instead. If you do that, we won't need to sort this array later. Something like 
```suggestion
                defferedUpdates[updateParams.lastUpdateID] = updateParams;
```